sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -542,6 +770,17 @@ pi_result piQueueGetInfo(pi_queue queue, pi_queue_info param_name,
   case PI_EXT_ONEAPI_QUEUE_INFO_EMPTY:
     // OpenCL doesn't provide API to check the status of the queue.
     return PI_ERROR_INVALID_VALUE;
+  case PI_QUEUE_INFO_DEVICE: {
+    if (queue2dev.find(queue) != queue2dev.end()) {
+      pi_device dev = queue2dev[queue];
+      if (param_value)
+        std::memcpy(param_value, &dev, sizeof(dev));
+      if (param_value_size_ret)
+        *param_value_size_ret = sizeof(pi_device);
+      return PI_SUCCESS;
+    }
+    [[fallthrough]];
```

Comment:
Is there ever a case where the queue should legitimately not be found in the queue2dev map?

I'm trying to decide whether it makes sense to fallthrough here, or whether it would be better to return an error.

Reply:
From spec: It is possible that a device(s) becomes unavailable after a context and command-queues that use this device(s) have been created and commands have been queued to command-queues. In this case the behavior of OpenCL API calls that use this context (and command-queues) are considered to be implementation-defined.
In this implementation, I just fall back onto CLErr that OpenCL returns in such cases. Hope that's ok.

Thanks